File: src/app/admin/employee/employee.ts
import { Component, OnInit } from '@angular/core';
import { CommonModule } from '@angular/common';
import { Router } from '@angular/router';
export interface EmployeeData {
  id: number;

  title: string;
  name: string;
  email: string;
  designation: string;
  status: 'Active' | 'Inactive';

  fatherName: string;
  motherName: string;
  gender: string;
  dateOfBirth: string;
  bloodGroup: string;
  phoneNumber: string;
  mobileNumber: string;
  personalEmail: string;
  nationality: string;
  maritalStatus: string;
  spouseName: string;
  aadhaarNumber: string;
  panNumber: string;
  religion: string;

  currentAddress1: string;
  currentAddress2: string;
  currentAddress3: string;
  currentCountry: string;
  currentState: string;
  currentCity: string;
  currentPincode: string;

  permanentAddress1: string;
  permanentAddress2: string;
  permanentAddress3: string;
  permanentCountry: string;
  permanentState: string;
  permanentCity: string;
  permanentPincode: string;

  emergencyContact: string;
  emergencyMobile: string;
  emergencyRelation: string;
}

@Component({
  selector: 'app-employee',
  standalone: true,
  imports: [CommonModule],
  templateUrl: './employee.html',
  styleUrl: './employee.css'
})
export class Employee implements OnInit {

  employees: EmployeeData[] = [];

  constructor(private router: Router) {}

  ngOnInit(): void {
    this.loadEmployees();
  }

  // =========================
  // LOAD EMPLOYEES
  // =========================

  loadEmployees(): void {
    this.employees = JSON.parse(
      localStorage.getItem('employees') || '[]'
    );
  }

  // =========================
  // OPEN NEW EMPLOYEE FORM
  // =========================

  openForm() {
  localStorage.removeItem('editingEmployee');
  this.router.navigate(['/admin/employee/new']);
}

  // =========================
  // EDIT EMPLOYEE
  // =========================

  editEmployee(employee: EmployeeData) {
  localStorage.setItem('editingEmployee', JSON.stringify(employee));
  this.router.navigate(['/admin/employee/new']);
}



  // =========================
  // VIEW EMPLOYEE
  // =========================

  selectedEmployee: EmployeeData | null = null;
showView = false;

viewEmployee(employee: EmployeeData) {
  this.selectedEmployee = employee;
  this.showView = true;
}

closeView() {
  this.showView = false;
  this.selectedEmployee = null;
}

  // =========================
  // DELETE EMPLOYEE
  // =========================

  deleteEmployee(employee: EmployeeData): void {

    const confirmed = confirm(
      `Are you sure you want to delete ${employee.name}?`
    );

    if (!confirmed) {
      return;
    }

    this.employees = this.employees.filter(
      item => item.id !== employee.id
    );

    localStorage.setItem(
      'employees',
      JSON.stringify(this.employees)
    );

    alert('Employee deleted successfully.');
  }
}

File: MyProject.API/Models/Employee.cs
namespace MyProject.API.Models;

public class Employee
{
    public int Id { get; set; }

    // Link to the staff login account
    public int? UserId { get; set; }
    public User? User { get; set; }

    // Personal details
    public string Title { get; set; } = string.Empty;
    public string EmployeeName { get; set; } = string.Empty;
    public string? FatherName { get; set; }
    public string? MotherName { get; set; }
    public string Gender { get; set; } = string.Empty;
    public string? DateOfBirth { get; set; }
    public string? BloodGroup { get; set; }

    // These create the staff login
    public string MobileNumber { get; set; } = string.Empty;
    public string PersonalEmail { get; set; } = string.Empty;

    public string? Nationality { get; set; }
    public string? MaritalStatus { get; set; }
    public string? SpouseName { get; set; }
    public string? AadhaarNumber { get; set; }
    public string? PanNumber { get; set; }
    public string? Religion { get; set; }

public string? Department { get; set; }
public string? Designation { get; set; }
public string? JoiningDate { get; set; }
    // Current address
    public string? CurrentAddress { get; set; }
    public string? CurrentCountry { get; set; }
    public string? CurrentState { get; set; }
    public string? CurrentCity { get; set; }
    public string? CurrentPincode { get; set; }

    // Permanent address
    public string? PermanentAddress { get; set; }
    public string? PermanentCountry { get; set; }
    public string? PermanentState { get; set; }
    public string? PermanentCity { get; set; }
    public string? PermanentPincode { get; set; }

    // Emergency contact
    public string? EmergencyContact { get; set; }
    public string? EmergencyMobile { get; set; }
    public string? EmergencyRelation { get; set; }
}